src/src.cpp

```diff
@@ -42,12 +94,21 @@ context::get_communicator()
 // communicator              //
 ///////////////////////////////
 
+communicator::communicator(impl_type* impl_)
+: m_impl{impl_}
+, m_pool{std::make_unique<boost::pool<>>(sizeof(detail::request_state), 128)}
+, m_schedule{std::make_unique<schedule>()}
+{
+    get_comm_set(m_impl->m_context).insert(std::make_pair(m_impl, m_schedule.get()));
```

Comment:
as I wrote above, it seems to me that `std::make_pair(m_impl, m_schedule.get())` is always unique. Why do you need a set here?

Reply:
because of the need to erase a given element when the communicator is destroyed, see above.